### ISSUE TITLE
how to convert my own slim model

### ISSUE THREAD

**Issue by jerryhouuu** [NVIDIA-Jetson/tf_to_trt_image_classification#14]
hello, i want to use tensorrt to speed up the facenet project. [davidsandberg/facenet
](https://github.com/davidsandberg/facenet)

I freeze the model into .pb tensorflow modle, but I am confused about how can I convert it into .plan.
and, I tried to modify the code to fit the model zoo' inception-resnet-v2 and re-train, but it could not be convert into an .uff model too.

**Comment by jaybdub-nv** [comment]
Hi jerryhouuu,

The scripts in this repository are only tested against the models listed.

It may help to watch the [TensorFlow to TensorRT on Jetson Webinar](http://info.nvidia.com/tensorflow-for-jetson-mar-2018-reg-page.html) for more information on how to optimize TensorFlow models with TensorRT, and better understand the known limitations.

That said, If you are able to provide the frozen graph of the model you are attempting to convert I may be able to provide more detailed insight.

**Comment by jerryhouuu** [comment]
@jaybdub-nv Thank you for your reply.
The error like below while I converted the frozen graph into uff model, could you give me some advise?
```
Converting to UFF graph
Warning: keep_dims is not supported, ignoring...
DEBUG: convert reshape to flatten node
Warning: No conversion function registered for layer: QueueDequeueUpToV2 yet.
Converting as custom op QueueDequeueUpToV2 batch_join
name: "batch_join"
op: "QueueDequeueUpToV2"
input: "batch_join/fifo_queue"
input: "batch_size"
attr {
  key: "component_types"
  value {
    list {
      type: DT_FLOAT
      type: DT_INT64
    }
  }
}
attr {
  key: "timeout_ms"
  value {
    i: -1
  }
}

```
```
        image_paths_placeholder = tf.placeholder(tf.string, shape=(None,1), name='image_paths')
        labels_placeholder = tf.placeholder(tf.int32, shape=(None,1), name='labels')
        input_queue = data_flow_ops.FIFOQueue(capacity=100000,
                                    dtypes=[tf.string, tf.int32],
                                    shapes=[(1,), (1,)],
                                    shared_name=None, name=None)
        enqueue_op = input_queue.enqueue_many([image_paths_placeholder, labels_placeholder], name='enqueue_op')
        
        nrof_preprocess_threads = 4
        images_and_labels = []
        for _ in range(nrof_preprocess_threads):
            filenames, label = input_queue.dequeue()
            images = []
            for filename in tf.unstack(filenames):
                file_contents = tf.read_file(filename)
                image = tf.image.decode_image(file_contents, channels=3)
                if args.random_rotate:
                    image = tf.py_func(facenet.random_rotate_image, [image], tf.uint8)
                if args.random_crop:
                    image = tf.random_crop(image, [args.image_size, args.image_size, 3])          
                else:
                    image = tf.image.resize_image_with_crop_or_pad(image, args.image_size, args.image_size)
                if args.random_flip:
                    image = tf.image.random_flip_left_right(image)
    
                #pylint: disable=no-member
                image.set_shape((args.image_size, args.image_size, 3))
                images.append(tf.image.per_image_standardization(image))
            images_and_labels.append([images, label])
            
    
        image_batch, label_batch = tf.train.batch_join(
            images_and_labels, batch_size=batch_size_placeholder, 
            shapes=[(args.image_size, args.image_size, 3), ()], enqueue_many=True,
            capacity=4 * nrof_preprocess_threads * args.batch_size,
            allow_smaller_final_batch=True)
```

**Comment by jerryhouuu** [comment]
@jaybdub-nv hello, could you give me some advise about the above error, Thank you

**Comment by xhzzhang** [comment]
@jerryhouuu hello, I got same issue as below:

ERROR: Parameter check failed at: Utils.cpp::reshapeWeights::71, condition: input.values != nullptr
ERROR: UFFParser: Parser error: res_aspp_g/decoder/resnet/bn_conv1/batch_normalization/FusedBatchNorm: reshape weights failed!
ERROR: sample_uff_ssd: Fail to parse

did you have any idea for this?
Thanks

**Comment by xhzzhang** [comment]
@jaybdub-nv 

When I convert pb to uff, got below issue:

xhz@xhz-omen:tf_to_trt_image_classification$ python scripts/convert_plan.py /home/xhz/Projects/models/xw_model/new/LsGan_model_test.pb /home/xhz/Projects/models/xw_model/LsGan_model.pb.plan input_img 360 640 sigmoid_logits 1 0 float
Using output node sigmoid_logits
Converting to UFF graph
WARNING: The UFF converter currently only supports 2D dilated convolutions
WARNING: The UFF converter currently only supports 2D dilated convolutions
WARNING: The UFF converter currently only supports 2D dilated convolutions
WARNING: The UFF converter currently only supports 2D dilated convolutions
WARNING: The UFF converter currently only supports 2D dilated convolutions
WARNING: The UFF converter currently only supports 2D dilated convolutions
WARNING: The UFF converter currently only supports 2D dilated convolutions
WARNING: The UFF converter currently only supports 2D dilated convolutions
WARNING: The UFF converter currently only supports 2D dilated convolutions
No. nodes: 570
UFF Output written to data/tmp.uff
UFFParser: parsing input_img
UFFParser: parsing res_aspp_g/decoder/resnet/conv1/weights
UFFParser: parsing res_aspp_g/decoder/resnet/conv1/conv1
UFFParser: Convolution: add Padding Layer to support asymmetric padding
UFFParser: Convolution: Left: 2
UFFParser: Convolution: Right: 3
UFFParser: Convolution: Top: 2
UFFParser: Convolution: Bottom: 3
UFFParser: parsing res_aspp_g/decoder/resnet/bn_conv1/BatchNorm/Const
UFFParser: parsing res_aspp_g/decoder/resnet/bn_conv1/BatchNorm/beta
UFFParser: parsing res_aspp_g/decoder/resnet/bn_conv1/BatchNorm/Const_1
UFFParser: parsing res_aspp_g/decoder/resnet/bn_conv1/BatchNorm/Const_2
UFFParser: parsing res_aspp_g/decoder/resnet/bn_conv1/BatchNorm/FusedBatchNorm
Parameter check failed at: Utils.cpp::reshapeWeights::71, condition: input.values != nullptr
UFFParser: Parser error: res_aspp_g/decoder/resnet/bn_conv1/BatchNorm/FusedBatchNorm: reshape weights failed!
Failed to parse UFF

**Comment by IamMegatron2025** [comment]
same error with `FusedBatchNorm`.  how to fix it?

**Comment by IamMegatron2025** [comment]
@jaybdub-nv  can you give us some advises?

**Comment by janchk** [comment]
Same error.

**Comment by deaffella** [comment]
> hello, i want to use tensorrt to speed up the facenet project. [davidsandberg/facenet ](https://github.com/davidsandberg/facenet)
> 
> I freeze the model into .pb tensorflow modle, but I am confused about how can I convert it into .plan.
> and, I tried to modify the code similar to the model zoo' inception-resnet-v2 and re-train, but it could not be convert into an .uff model too.

Hi!
I want to convert facenet model to trt but it seems unreacheble.
Have you resolve this problem?

**Comment by 250zhanghu** [comment]
> @jerryhouuu thanks for sharing.
> 

thanks for your advice. and i will try it
